lib/common/ipc_attrd.c

```diff
@@ -181,6 +196,74 @@ pcmk__attrd_api_query(pcmk_ipc_api_t *api, const char *node, const char *name,
     return rc;
 }
 
+int
+pcmk__attrd_api_refresh(pcmk_ipc_api_t *api, const char *node)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *request = NULL;
+    const char *display_host = (node ? node : "localhost");
+
+    request = create_attrd_op(NULL);
+
+    crm_xml_add(request, PCMK__XA_TASK, PCMK__ATTRD_CMD_REFRESH);
+    crm_xml_add(request, PCMK__XA_ATTR_NODE_NAME, node);
+
+    rc = send_attrd_request(api, request);
+    free_xml(request);
+
+    crm_debug("Asked pacemaker-attrd to refresh %s: %s (%d)",
+              display_host, pcmk_rc_str(rc), rc);
+
+    return rc;
+}
+
+int
+pcmk__attrd_api_update(pcmk_ipc_api_t *api, const char *node, const char *name,
+                       const char *value, const char *dampen, const char *set,
+                       const char *user_name, uint32_t options)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *request = NULL;
+    const char *display_host = (node ? node : "localhost");
+
+    if (name == NULL) {
+        return pcmk__attrd_api_refresh(api, node);
+    }
```

Comment:
I think this should return EINVAL instead -- with the new API there's no reason to make update do anything but update.

Reply:
Fixed.  I don't think anything was passing NULL when it meant refresh, but I guess there's only one way to find out.